cromwell-backend/src/main/scala/cromwell/backend/model/TaskDescriptor.scala

```diff
@@ -0,0 +1,24 @@
+package cromwell.backend.model
+
+import wdl4s.TaskOutput
+import wdl4s.command.CommandPart
+import wdl4s.values.WdlValue
+
+/**
+  * Represents basic data needed to execute a task in a defined backend.
+ *
+  * @param name Task name.
+  * @param user User who want to execute the task.
+  * @param commandTemplate Arguments for the execution.
+  * @param workingDir Task working directory.
+  * @param outputs Outputs generated by the execution.
+  * @param runtimeAttributes Configuration needed at runtime in order to execute a task.
+  */
+case class TaskDescriptor(name: String,
+                          user: String,
+                          commandTemplate: Seq[CommandPart], //TODO: Should be Wdl Agnostic
```

Comment:
It's usually not necessary I just wasn't sure who it was directed at.

We have sketched out a roadmap but it hasn't gone past that point simply due to prioritization. After the dust settles on this if that's something you all were still interested in helping out with (or if the priorities set for us aren't fast enough) we should chat a bit. 

At a high level it involves combining the concepts from the java CWL sdk and wdl4s to derive a fairly generic workflow object model that cromwell will use. The only part which will actually be intellectually difficult is figuring out how to handle expressions.